CONTRIBUTING.md

```diff
@@ -42,3 +42,45 @@ test case. To make this process easier, we have prepared one basic
 Maintenance branches are how we manage the different supported point releases
 of RSpec. As such, while they might look like good candidates to merge into
 main, please do not open pull requests to merge them.
+
+## Working on multiple RSpec gems at the same time
+
+RSpec is composed of multiple gems (`rspec-core`, `rspec-mocks`, etc). Sometimes you have
+to work on a combination of them at the same time. When submitting your code for review,
+we ask that you get a passing build (green CI). If you are working across the repositories,
+please add a commit that temporarily pins your PR to the right branch of the other repository
+you depend on. For example, if we wanted a change in `rspec-expectations` that relied on a
+change for on `rspec-mocks`. We add a commit with the title:
+
+>[WIP] Use rspec-mocks with "custom-failure-message" branch
+
+And content:
+
+```diff
+diff --git a/Gemfile b/Gemfile
+
+-%w[rspec rspec-core rspec-mocks rspec-support].each do |lib|
++%w[rspec rspec-core rspec-support].each do |lib|
+   library_path = File.expand_path("../../#{lib}", __FILE__)
+   if File.exist?(library_path) && !ENV['USE_GIT_REPOS']
+     gem lib, :path => library_path
+@@ -11,6 +11,7 @@ branch = File.read(File.expand_path("../maintenance-branch", __FILE__)).chomp
+     gem lib, :git => "https://github.com/rspec/#{lib}.git", :branch => branch
+   end
+ end
++gem 'rspec-mocks', :git => "https://github.com/rspec/rspec-mocks.git", :branch => "custom-failure-message"
+```
+
+In general the process is:
+1. Create PRs explaining what you are trying to achieve.
+2. Pin the repositories to each other.
+3. Check they pass (go green).
+4. Await review if appropriate.
+5. Remove the commit from one. We will merge ignoring the failure.
+6. Remove the commit from the other, check it passes with the other commit now on main.
+7. Merge the other.
+8. Rebuild the other main to check it still passes.
```

Comment:
Can we trigger "main" by hand?
"Rebuild" sounds like "you should rebuild", while it's probably not possible for contributors. Do you think it makes sense to rephrase this to something like "we will trigger builds for the main branch of affected repositories to check if everything is in order"?